examples/systemcalls/systemcalls.c

```diff
@@ -92,7 +120,46 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *   
 */
+    pid_t pid;
+    int status;
+
+    pid = fork();
+    if (pid == -1) {
+        perror("fork");
+        return false;
+    } 
+    else if (pid == 0) {
+        int fd;
+        if ((fd = open(outputfile, O_WRONLY | O_CREAT | O_TRUNC, 0777)) == -1) {
+            perror("open");
+            return false;
+        }
+
+        if (dup2(fd, STDOUT_FILENO) == -1) {
+            perror("dup2");
+            return false;
+        }
 
+        close(fd);
+        
+        if (execv(command[0], &command[0]) == -1) {
+            perror("execv");
+            return false;
+        }
```

Comment:
@vido2373 consider ways to combine some of the logic here and in the `do_exec` function above.